### DISCUSSION TITLE
Failed to arm via QGC

### DISCUSSION THREAD

**Discussion by LoreBene99** [RosettaDrone/rosettadrone#164] [Q&A]
Hi to everybody, sorry to bother you. Recently I'm working on QGC using rosettadrone for the purpose of waypoint mission (using the DJI Mini 2 drone) and I'm facing some problems. Basically, when I try to take off the drone and when I try to start a waypoint mission from QGC on my laptop and rosettadrone opened on my smartphone, nothing happen, even if the GPS is working and the drone is connected to the RC and QGC. Instead, if I manually take off the drone and then I press the "Land" action from QGC, it works properly. How can I fix this problem? Is this project suppose to work for waypoint mission using DJI MINI 2. In the end, even if the waypoint mission is not supported, at least the "Take off" action should work, right? Thanks in advance for your help!

**Comment by kripper** [comment]
Should work fine with mini 2.
Check the logs to see what's going on.

**Reply by LoreBene99** [reply]
Ok thank you. I tried to take off the drone without arming manually the drone. So I read the QGC Log (Application settings - Console) : [E] at :0 - "QGeoTileRequestManager: Failed to fetch tile (1100772,759299,21) 5 times, giving up. Last error message was: 'Bing tile above zoom level'". I also tried to arm manually the vehicle and start a casual mission (Take off) --> the upload is succcesfull and QGC mode switch to autopilot, but the mission did not start and on the QGC Log this appear : [D] at D:\a\qgroundcontrol\qgroundcontrol\src\MissionManager\MissionController.cc:1272 - "_recalcFlightPathSegments homePositionValid true".  Whereas on Rosetta I enabled the "Log MAVlink messages" button, but when I tried to download the Log directly from the settings, It did not download anything (I do not know precisely where the download is located). Thank you for your time!

**Comment by kripper** [comment]
Unrelated messages.
Debug the take-off procedure.
Make sure to check Rosetta logs, not QGC.

**Comment by The1only** [comment]
Did you unlock/arm Rosettadrone befor you did anything? If not the you can not takeoff from QGC.Best regardTerje Nilsen9Tek ASNorwayOn 22 May 2023, at 18:12, Christopher Pereira ***@***.***> wrote:﻿
Unrelated messages.
Debug the take-off procedure.
Make sure to check Rosetta logs, not QGC.

**Reply by LoreBene99** [reply]
What do you mean by unlock/arm? The only thing that I did in Rosetta is to disable the Safety mode (the green lock on the up-right).
In fact when I tried to take off on QGC I didn't receive the message "Disable safety mode". Thank you.

**Comment by kripper** [comment]
QGC says "disable safety mode" in this case.

**Reply by LoreBene99** [reply]
I disabled the safety mode and when I try to arm the drone on QGC (pressing on the "Ready to fly" button) an error pop out : MAV_CMD_COMPONENT_ARM_DISARM command denied. Thank you.

**Reply by andreamanera** [reply]
I am working with @LoreBene99 on this project, we found from the read me that the ARM button in QGC does not work, anyway we still have the mentioned issues with the takeoff and the start mission, basically we don't get any error neither from QGC  or Rosetta, it seems like for QGC everything is working fine, but in reality nothing happen, now we would like to take a look to the Rosetta log files to try to find the problem (maybe Rosetta does not receive the command from QGC(?)), but when we press the download Log button on  the phone nothing happen, do you have some suggestion? (maybe it download the log to a particular path on my smartphone)

**Comment by kripper** [comment]
I remember there was some problem with the arm command the first time it is triggered from QGC. Maybe a timeout or some previous checking performed by QGC that fails.

Have you tried changing mode to AUTO in QGC after uploading the mission (top left)?
It uses to resume/start the mission after the first arm command failed.

**Reply by kripper** [reply]
> I remember there was some problem with the arm command the first time it is triggered from QGC. Maybe a timeout or some previous checking performed by QGC that fails.

Reported here:
https://github.com/mavlink/MAVSDK/issues/1990

**Comment by kripper** [comment]
I use Android Studio to see and filter the logs. Also useful for debugging.

See [important recomendations](https://docs.google.com/document/d/1wR56Mvd2OsT82kxMOD7f_3-c6yJGBtLaT8LuVrmC5wI/edit?usp=sharing)

**Comment by kripper** [comment]
DJIs don't really do a "arm", but Rosetta supports the MAVLink message and should respond with an ACK.

I would recommend you to check (debug) if the `MAV_CMD_COMPONENT_ARM_DISARM` **deny** message is really sent by Rosetta to QGC and why.
